### PR TITLE
Start reworking arch docs to provide an overview of all the models an…

### PR DESCRIPTION
**Pull request by VictorLowther**
…d what they do

### CODE REVIEW

File: doc/arch/design.rst [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+.. Digital Rebar Provision documentation under Digital Rebar master license
+.. index::
+  pair: Digital Rebar Provision; Architecture Design
+
+Architecture Overview
+---------------------
+
+Architecturally, dr-provision is split into several different packages:
+
+- backend is responsible for making sure that all the data is valid
```
Comment (zehicle) [inline doc/arch/design.rst]: if would be handy to bold these items or use a definition list.

File: doc/arch/models.rst [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+Model Metadata
+--------------
+
+Virtually every model contains some embedded metadata, available under
+the **Meta** field.  This data, which consists of a map of string ->
+string pairs, is ignored by the dr-provision server.  The most common
+metadata fields you will see are:
+
+icon
+  The icon that the UX will use to display instances of this model.
```
Comment (zehicle) [inline doc/arch/models.rst]: users can pick icons from: http://fontawesome.io/icons/

File: doc/arch/sledgehammer.rst [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+OS image.  It is based on the latest CentOS 7.x live CD, with any
+unneeded content stripped out, tools we usually use preinstalled, and
+rebacled to me more space and network bandwidth efficient.
+https://github.com/digitalrebar/sledgehammer is the Git repo that
+contains the code needed to build Sledgehammer.
+
+Sledgehammer Build
+------------------
+
+RackN provides pre-built Sledgehammer images that should function
```
Comment (zehicle) [inline doc/arch/sledgehammer.rst]: would be better to say "Digital Rebar" here instead of RackN

Comment (galthaus) [reply doc/arch/sledgehammer.rst]: Hmm - Digital Rebar doesn't provide those. RackN provides prebuilt Sledgehammers for use by the Digital Rebar community.

**Pull request merged**